contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -719,6 +725,61 @@ plsql_TriggerRecursiveCheck(ResultRelInfo *resultRelInfo)
 	return false;
 }
 
+/**
+ * Hook function to skip rewriting VIEW with base table if the VIEW has an instead of trigger
+ * Checks if view have an INSTEAD OF trigger at statement level
+ * If it does, we don't want to treat it as auto-updatable.
+ * This function also does error checking for recursive triggers
+ * Reference - src/backend/rewrite/rewriteHandler.c view_has_instead_trigger
+ */
+static bool
+pltsql_bbfViewHasInsteadofTrigger(Relation view, CmdType event)
+{
+	TriggerDesc *trigDesc = view->trigdesc;
+	if (trigDesc && triggerInvocationSequence)
+	{
+		int i;
+		for (i = 0; i < trigDesc->numtriggers; i++)
+		{
+			Trigger *trigger = &trigDesc->triggers[i];
+			Oid current_tgoid = trigger->tgoid;
+			Oid prev_tgoid = InvalidOid;
+			prev_tgoid = lfirst_oid(list_tail(triggerInvocationSequence));
+			if (prev_tgoid == current_tgoid)
+			{
+				return false; /** Loop trigger call by itself*/
+			}
+			else if (list_length(triggerInvocationSequence) > 32 || list_member_oid(triggerInvocationSequence, current_tgoid))
```

Comment:
Where did you get this number ? 
Can this number be adjusted in Sql server ?

Reply:
https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/triggers/create-nested-triggers.md
It says DML and DDL triggers can be nested up to 32 levels. I dont think this number(level 32) can be changed. It can only change whether to allow triggers of that type or not.  
Also sp_configure and alter database is not yet supported in Babelfish, through which settings related to nesting and recursion can be changed.

Above document has links for same
To set the nested triggers option
[sp_configure (Transact-SQL)](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/system-stored-procedures/sp-configure-transact-sql.md)
To set the RECURSIVE_TRIGGERS database option

[ALTER DATABASE SET Options (Transact-SQL)](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/statements/alter-database-transact-sql-set-options.md)